src/Sylius/Component/Order/Modifier/OrderItemQuantityModifier.php

```diff
@@ -42,6 +42,8 @@ public function modify(OrderItemInterface $orderItem, $targetQuantity)
             return;
         }
 
+        $targetQuantity = min($targetQuantity, 9999);
```

Comment:
Let's decorate it and use hardcoded value. No need to extract this value as a parameter, because it's more like a technical limit than business rule (the service can be overridden anyway, but we don't need to maintain the parameter).